cmdx.py

```diff
@@ -2627,7 +2681,24 @@ def __str__(self):
         return str(self.read())
 
     def __repr__(self):
-        return str(self.read())
+        cls_name = '{}.{}'.format(__name__, self.__class__.__name__)
+        msg = '{}("{}", "{}")'.format(cls_name,
+                                      self.node().name(),
+                                      self.name())
+
+        if not all((self.isCompound, self.isArray)):
```

Comment:
Do you mean `any()`? This means it can be an array, and it can be a compound, but it cannot be both at the same time

Reply:
`all` is correct. `any` would exclude any attribute that isn’t an array or compound.

The idea is that it can display an array _(in the case of `worldMatrix`, for example)_ and a compound _(in the case of `translate`)_. But, when dealing with a Compound that is an Array, that’s when it seems to start to become unclear what sort of data will be displayed, namely the example of `bs['input']` mentioned previously in this PR.

Reply:
> But, when dealing with a Compound that is an Array, that’s when it seems to start to become unclear what sort of data will be displayed, namely the example of bs['input'] mentioned previously in this PR

Ahaa, yes in that case. Good. Then this looks good I think! Well done, let's get this show on the road! 👏